src/daemon/daemon.cpp

```diff
@@ -2607,6 +2640,12 @@ try
             {
                 const auto& instance_name = vm_it->first;
 
+                if (vm_it->second->current_state() == VirtualMachine::State::unavailable)
+                {
+                    mpl::log(mpl::Level::info, instance_name, "Ignoring delete since instance is unavailable.");
```

Comment:
Ouch, this looks a little radical. Suppose that are 5 running instances in a zone before you disable it. You then decide to delete some of them. Do we require people to restart all of them first?